docs/editing-components.md

```diff
@@ -0,0 +1,376 @@
+            </h2>
+        </header>
+    {% endif %}
+
+    {{ value.paragraph | safe }}
+
+    {% if value.links %}
+        <ul class="m-list m-list__links">
+        {% for link in value.links %}
+            <li class="m-list_item">
+                <a href="{{ link.url }}" class="m-list_link">{{ link.text }}</a>
+            </li>
+        {% endfor %}
+        </ul>
+    {% endif %}
+</div>
+```
+
+When Wagtail renders a block,
+it includes the values of its sub-blocks in an object named `value`.
+Above, you can see where the sub-blocks are output with
+[Jinja2 expression tags](http://jinja.pocoo.org/docs/2.10/templates/#expressions).
+In particular, note how the `links` `ListBlock` is iterated over,
+and the values of _its_ `Hyperlink` child blocks are output.
+
+That's about as simple an example as it gets, but block templates can get much
+more complex when they have lots of child blocks and grandchild blocks.
+
+
+### Adding some style
+
+If a component needs any custom styling not already provided
+by Capital Framework or cfgov-refresh,
+you can it by creating a new Less file for the component.
+This file should live in `cfgov/unprocessed/css/<atoms|molecules|organisms>`.
+Choose the deepest folder according to the atomic rank of the component.
+Continuing the `RelatedContent` example, if it needed its own styles,
+it would live at `cfgov/unprocessed/css/molecules/related-content.less`.
+
+Newly-created Less files need to be imported into the project's master
+`main.less` file, located at `cfgov/unprocessed/css/main.less`.
+Please place them in the appropriate section for their atomic rank.
+
+Because cfgov-refresh uses `main.less` to build a single CSS file
+for the entire project, it is not necessary
+to tell the Python model anything about a component-specific stylesheet.
+That is _not_ the case with JavaScript, as we will see in the next section.
+
+
+### Adding some JavaScript
+
+Each atomic component may optionally be given a `Media` class that can
+list one or more JavaScript files that should be loaded when using it.
+When a page is requested via the browser,
+[code contained in `base.html`](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/jinja2/v1/_layouts/base.html#L100-L110)
+will loop all atomic components for the requested page and
+load the appropriate atomic JavaScript bundles.
+
+Here is how one would add the `Media` class to our `RelatedContent` example:
+
+```python
+class RelatedContent(blocks.StructBlock):
+    …  # see first example on this page
+
+    class Media:
+        js = ['related-content.js']
+```
+
+(The `related-content.js` file would need to be placed in
+`cfgov/unprocessed/js/molecules/`.)
+
+This will load the `related-content.js` script on any page
+that includes the `RelatedContent` molecule in one of its StreamFields.
+
+
+
+
+## How-to guides
+
+
+### Adding, editing, or removing fields
+
+#### Adding a field
+
+1. Locate the Python class of the component you want to add a field to in
```

Comment:
See above re: generalizing the location of blocks.

Reply:
Seconded.

Reply:
Addressed in a47039c